numpy/core/include/numpy/experimental_dtype_api.h

```diff
@@ -370,6 +370,30 @@ typedef int (get_reduction_initial_function)(
 #define NPY_DT_setitem 7
 #define NPY_DT_getitem 8
 
+// These slots will be deprecated and replaced eventually
+#define NPY_DT_PyArray_ArrFuncs_getitem 0 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_setitem 1 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_copyswapn 2 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_copyswap 3 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_compare 4 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_argmax 5 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_dotfunc 6 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_scanfunc 7 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_fromstr 8 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_nonzero 9 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_fill 10 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_fillwithscalar 11 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_sort 12 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_argsort 13 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_castdict 14 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_scalarkind 15 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_cancastscalarkindto 16 + (1 << 10)
+#define NPY_DT_PyArray_ArrFuncs_cancastto 17 + (1 << 10)
```

Comment:
All cast related slots should go IMO, maybe just comment them out?  Some of the others are also unused I believe, e.g. fastclip is a ufunc.
`getitem`/`setitem` could stay (as a performance optimization), but have a default (maybe comment?).

Reply:
Sounds good - they've been commented out and an exception is raised if someone tries to use them. I've also disabled the deprecated `fastclip`, `fastputmask`, and `fasttake` here too.